providers/chefinfra/chefinfra.go

```diff
@@ -16,12 +16,101 @@ limitations under the License.
 
 package chefinfra
 
-import "fmt"
+import (
+	"crypto/sha256"
+	"encoding/json"
+	"errors"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"strings"
+)
 
 // func executePolicyArchive() {}
 // func passInAttrAsJson() {}
 
-// Execute execs
-func Execute() {
-	fmt.Println("Infra!")
+type chefInfraDefinition struct {
+	args           []string
```

Comment:
We should probably update in the instructions about running that `--local-mode` and `--json-attributes` are passed automatically.  Assuming it would bomb if someone passed them into the chef args.

Reply:
you wouldn't be able to pass args in.  They aren't Unmarshalled, and even if they were, they get overwritten.  They are completely internal to the program.